### DISCUSSION TITLE
Cannot Delete Project Cards from undefined firebaseKey

### DISCUSSION THREAD

**Discussion by robertncole1** [nss-evening-cohort-14/evening-client-side#250] [Help Me!] [answered]
I have searched far and wide through my code to find out why my data will not delete from firebase. I believe it has something to do with firebase key not being found when I call my deleteProject function. However, I am passing the props down, and everything looks good. I cannot figure it out.  I also am getting no error that points me in the right direction. However, I did put in a console warn to check and make sure my handleProjectsButton was working and it is when I press delete.

What have I googled?
What haven't I googled. I have searched why projects won't delete from firebase. I have tried searching proptypes and where they should be passed. And a lot more.

Link to my repo:
https://github.com/robertncole1/personal-bio-site

deleteProject function call:
https://github.com/robertncole1/personal-bio-site/blob/main/src/App/components/Projects.js

axios calls:
https://github.com/robertncole1/personal-bio-site/blob/main/src/helpers/data/projectData.js

Any advice or help would be appreciated!

**Comment by TrinityChristiana** [comment, marked as answer]
Looks like the needed firebaseKey does not need to come from a prop, it needs to come directly from the [project in the  `.map`](https://github.com/robertncole1/personal-bio-site/blob/main/src/App/components/Projects.js#L42)
So it will need to be [passed down to `editView`](https://github.com/robertncole1/personal-bio-site/blob/main/src/App/components/Projects.js#L52), then [passed down to `handleProjectsButton `](https://github.com/robertncole1/personal-bio-site/blob/main/src/App/components/Projects.js#L37)

**Reply by robertncole1** [reply]
Thank you! Once I passed down firebaseKey to my editView function and the handleProjectsButton, it worked! Fun fact, I also realized after that I also need to pass the firebaseKey to my edit button because that was also inside my editView function.

**Reply by TrinityChristiana** [reply]
Yes! Way to apply what you've learned!! 🙌🏽
